### ISSUE TITLE
Problems installing on ubuntu 24.04

### ISSUE THREAD

**Issue by castroguifetz** [MysticRyuujin/guac-install#274]
I'm installing it on Ubuntu 24.04 but I ran into some problems, the first was the tomcat version (we had to update to version 10 in the script), after that it was installed without any additional problems, but on time When using it, it does not access the login page, we tried different ways, we checked the files in the /var/lib/tomcat10/webapps folders and they are all correct, but they do not open, it keeps giving a 404 error, we created an hmtl file for testing and the Even though it opened normally, I would like your help to solve this problem

**Comment by itiligent** [comment]
See this much more recent jumphost build script which will fully work on Ubuntu 24.04:
[https://github.com/itiligent/Guacamole-Install](https://github.com/itiligent/Guacamole-Install)

FYI Guacamole is not currently compatible with Tomcat 10 and Guac devs dont have a timeline for when this will be added. In the dev wiki there is discussion that notes the Tomcat 9 will be under support for another year, so it may be that long before there is a solution.    BTW This repo is pretty much out of date now and its not maintained. Its a bit of a fluke that it mostly can still work on some Ubuntu, but on Debian 11 or 12 it will generally fail.

**Comment by TayloredSoftware** [comment]
That is correct. Tomcat 10 is not currently compatible but a PR has been started at [](https://github.com/MysticRyuujin/guac-install/issues/274) and is getting very close. May see that compatibility in the next two versions. For the time being you would need to manually install Tomcat 9 and create a systemd entry. Just completed an install on 24.04 using these steps without issue. 

wget [](https://downloads.apache.org/tomcat/tomcat-9/v9.0.90/bin/apache-tomcat-9.0.90.tar.gz)
sudo mkdir /opt/tomcat
sudo tar xzvf apache-tomcat-9.0.90.tar.gz -C /opt/tomcat --strip-components=1
sudo groupadd tomcat
sudo useradd -s /bin/false -g tomcat -d /opt/tomcat tomcat
sudo chown -R tomcat:tomcat /opt/tomcat
sudo chmod -R 755 /opt/tomcat
sudo nano /etc/systemd/system/tomcat9.service
[Unit]
Description=Apache Tomcat 9
After=network.target

[Service]
Type=forking

Environment=JAVA_HOME=_**/usr/lib/jvm/ ( Need to find your location)**_
Environment=CATALINA_PID=/opt/tomcat/temp/tomcat.pid
Environment=CATALINA_HOME=/opt/tomcat
Environment=CATALINA_BASE=/opt/tomcat
Environment='CATALINA_OPTS=-Xms512M -Xmx1024M -server -XX:+UseParallelGC'
Environment='JAVA_OPTS=-Djava.awt.headless=true -Djava.security.egd=file:/dev/./urandom'

ExecStart=/opt/tomcat/bin/startup.sh
ExecStop=/opt/tomcat/bin/shutdown.sh

User=tomcat
Group=tomcat
UMask=0007
RestartSec=10
Restart=always

[Install]
WantedBy=multi-user.target
sudo chown -R tomcat:tomcat /opt/tomcat/logs
sudo chmod -R 755 /opt/tomcat/logs
sudo chown -R tomcat:tomcat /opt/tomcat
sudo chmod -R 755 /opt/tomcat
sudo systemctl daemon-reload
sudo systemctl start tomcat9
sudo systemctl enable tomcat9
sudo systemctl status tomcat9

**Comment by cowbe0x004** [comment]
Also 24.04 and got the can't find tomcat9 package error. You can let the script install tomcat9 after install this repo,
```
add-apt-repository -y -s "deb http://archive.ubuntu.com/ubuntu/ jammy main universe"
```

**Comment by PowerPCFan** [comment]
I'm on 24.04 LTS or 24.10 (I honestly don't remember lol) and it was also failing at that point, I'm still having some other unrelated issues with installation and MySQL but to fix the Tomcat issue I added the Jammy Jellyfish Universe repo to apt, after finding out that compatibility with Apache Tomcat 10 was broken/limited

**Comment by PowerPCFan** [comment]
Turns out the MySQL problems were only happening on 24.04, I decided to try Jammy Jellyfish 22.04 LTS itself and it worked fine - use that!!

**Comment by azothdm** [comment]
Or you can do it:
echo "deb http://archive.ubuntu.com/ubuntu/ jammy universe" | tee /etc/apt/sources.list.d/jammy.list
apt update

and run the script again

**Comment by beckchin28** [comment]
CVE-2025-24813 in Tomcat 9 and Tomcat 10 is only patched in the Pro version of Ubuntu. It's better to install Tomcat manually.
